app/config/services/platform.xml

```diff
@@ -427,11 +427,33 @@
             <argument type="service" id="cache.app"/>
         </service>
 
-        <service id="app.search.search_results_provider" class="AppBundle\Search\SearchResultsProvider">
+        <service id="app.search.search_results_providers_manager" class="AppBundle\Search\SearchResultsProvidersManager">
+            <call method="addProvider">
```

Comment:
After an upgrade to 3.4 you could use:
```
<call method="addProviders">
    <argument type="tagged" tag="app.search_provider" />
</call>
```
And auto tag providers below thanks to their interface if any.

Reply:
I would love that 👍